ui/ducks/send/send.js

```diff
@@ -705,12 +728,85 @@ const slice = createSlice({
       slice.caseReducers.calculateGasTotal(state);
     },
     /**
-     * sets the provided gasPrice in state and then recomputes the gasTotal
+     * Sets the appropriate gas fees in state and determines and sets the
+     * appropriate transactionType based on gas fee fields received.
      */
-    updateGasPrice: (state, action) => {
-      state.gas.gasPrice = addHexPrefix(action.payload);
+    updateGasFees: (state, action) => {
+      if (
+        action.payload.transactionType === TRANSACTION_ENVELOPE_TYPES.FEE_MARKET
+      ) {
+        state.gas.maxFeePerGas = addHexPrefix(action.payload.maxFeePerGas);
+        state.gas.maxPriorityFeePerGas = addHexPrefix(
+          action.payload.maxPriorityFeePerGas,
+        );
+        state.transactionType = TRANSACTION_ENVELOPE_TYPES.FEE_MARKET;
+      } else {
```

Comment:
I don't understand why this says "Until we remove the old UI we **don't** want to update gasPrice if the user has changed the field value." It seems inconsistent with the logic below, which updates gasPrice... I should come back to this

- [ ] Understand this

Reply:
Okay, what is meant here is: "Until we remove the old UI we don't want to _automatically_ update gasPrice if the user has already manually changed the field value"